### PR TITLE
Facility / region export

### PR DESCRIPTION
**Pull request by rsanheim**
**Story card:** [chch5183](https://app.shortcut.com/simpledotorg/story/5183/email-draft-region-structure-to-bd-team?vc_group_by=day&ct_workflow=all&cf_workflow=500000031)

## Because

We want a CSV export of all facilities with their related regions to show the BD team before we run the import in prod.

## This addresses

adds a `.csv` format to the admin FacilitiesController, as well as a small CSV export class to handle the values and the sorting.

## Test instructions

Go to a review site or localhost w/ this PR and access `/admin/facilities.csv`.

### CODE REVIEW

File: lib/data_scripts/update_bangladesh_regions_script.rb [hunk 1]
```diff
@@ -27,6 +27,7 @@ def initialize(dry_run: true, csv_path: DEFAULT_CSV_PATH)
 
   def call
     destroy_empty_facilities
+    fix_incorrect_regions
```
Comment (rsanheim) [inline lib/data_scripts/update_bangladesh_regions_script.rb:30]: There may be more of these, but wanted to add a little example of how we can fix these up.

File: spec/lib/data_scripts/update_bangladesh_regions_script_spec.rb [hunk 2]
```diff
@@ -42,25 +44,25 @@
...
+        expect {
+          result = described_class.call(dry_run: false, csv_path: test_csv_path)
+          result[:errors].each { |type, count| expect(count).to eq(0) }
+          expect(result[:created][:facility_regions]).to eq(45)
+          expect(result[:created][:facilities]).to eq(45)
+        }.to change { Region.count }.by(66)
+          .and change { Region.facility_regions.count }.by(45)
+          .and change { Region.district_regions.count }.by(6)
+          .and change { Facility.count }.by(45)
+          .and change { FacilityGroup.count }.by(6)
```
Comment (rsanheim) [inline spec/lib/data_scripts/update_bangladesh_regions_script_spec.rb:65]: Making this test more maintable.

**Review by rsanheim — COMMENTED** [review]
notes

File: app/services/facility_region_csv.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+class FacilityRegionCsv
```
Comment (timcheadle) [inline app/services/facility_region_csv.rb:1]: Should this go into `app/exporters` like our other CSV classes?

**Review by kpethtel — APPROVED** [review]
LGTM!

File: app/services/facility_region_csv.rb (hunk 3 above)
Comment (rsanheim) [reply app/services/facility_region_csv.rb:1]: Right now I think its still pretty specific to this BD purpose -- if we end up generalizing it I think it should definitely move there.

**Pull request merged**